src/index.ts

```diff
@@ -47,20 +45,19 @@ export default class AutoCheckElement extends HTMLElement {
     }
   }
 
-  static get observedAttributes(): Array<string> {
+  static get observedAttributes(): string[] {
     return ['required']
   }
 
-  get input(): ?HTMLInputElement {
-    const input = this.querySelector('input')
-    return input instanceof HTMLInputElement ? input : null
+  get input(): HTMLInputElement | null {
+    return this.querySelector('input')
   }
 
   get src(): string {
     const src = this.getAttribute('src')
     if (!src) return ''
 
-    const link = this.ownerDocument.createElement('a')
+    const link = this.ownerDocument!.createElement('a')
```

Comment:
I just cut version 4.0.0 of `eslint-plugin-github`. If you update to that version, the ESLint violation here should go away.